SmartDeviceLink/SDLProtocol.m

```diff
@@ -507,66 +566,98 @@ - (void)handleHeartbeatForSession:(Byte)session {
     if (msg.header.serviceType == SDLServiceTypeControl) {
         [self sdl_processSecurityMessage:msg];
         return;
     }
 
-    for (id<SDLProtocolListener> listener in self.protocolDelegateTable.allObjects) {
+    NSArray<id<SDLProtocolListener>> *listeners = [self sdl_getProtocolListeners];
+    for (id<SDLProtocolListener> listener in listeners) {
         if ([listener respondsToSelector:@selector(onProtocolMessageReceived:)]) {
             [listener onProtocolMessageReceived:msg];
         }
     }
 }
 
 - (void)onProtocolOpened {
-    for (id<SDLProtocolListener> listener in self.protocolDelegateTable.allObjects) {
+    NSArray<id<SDLProtocolListener>> *listeners = [self sdl_getProtocolListeners];
+    for (id<SDLProtocolListener> listener in listeners) {
         if ([listener respondsToSelector:@selector(onProtocolOpened)]) {
             [listener onProtocolOpened];
         }
     }
 }
 
 - (void)onProtocolClosed {
-    for (id<SDLProtocolListener> listener in self.protocolDelegateTable.allObjects) {
+    NSArray<id<SDLProtocolListener>> *listeners = [self sdl_getProtocolListeners];
+    for (id<SDLProtocolListener> listener in listeners) {
         if ([listener respondsToSelector:@selector(onProtocolClosed)]) {
             [listener onProtocolClosed];
         }
     }
 }
 
 - (void)onError:(NSString *)info exception:(NSException *)e {
-    for (id<SDLProtocolListener> listener in self.protocolDelegateTable.allObjects) {
+    NSArray<id<SDLProtocolListener>> *listeners = [self sdl_getProtocolListeners];
+    for (id<SDLProtocolListener> listener in listeners) {
         if ([listener respondsToSelector:@selector(onError:exception:)]) {
             [listener onError:info exception:e];
         }
     }
 }
 
 - (void)sdl_logControlNAKPayload:(SDLProtocolMessage *)nakMessage {
-    if (nakMessage.header.version >= 5) {
-        SDLControlFramePayloadNak *endServiceNakPayload = [[SDLControlFramePayloadNak alloc] initWithData:nakMessage.payload];
-        NSArray<NSString *> *rejectedParams = endServiceNakPayload.rejectedParams;
-        if (rejectedParams.count > 0) {
-            SDLLogE(@"Start Service NAK'd, service type: %@, rejectedParams: %@", @(nakMessage.header.serviceType), rejectedParams);
-        }
+    switch (nakMessage.header.frameData) {
+        case SDLFrameInfoStartServiceNACK: // fallthrough
+        case SDLFrameInfoEndServiceNACK:
+            if (nakMessage.header.version >= 5) {
+                SDLControlFramePayloadNak *endServiceNakPayload = [[SDLControlFramePayloadNak alloc] initWithData:nakMessage.payload];
+                NSArray<NSString *> *rejectedParams = endServiceNakPayload.rejectedParams;
+                if (rejectedParams.count > 0) {
+                    SDLLogE(@"Start Service NAK'd, service type: %@, rejectedParams: %@", @(nakMessage.header.serviceType), rejectedParams);
+                }
+            }
+            break;
```

Comment:
Move up a line

Reply:
I remember your preference is to add `{ ... }` in each `case` block, so I updated the code into such style. Please refer to commit b4455699e2f589b9e46db24e96e86015575f6442.